src/app/demo/demo.component.spec.ts

```diff
@@ -239,3 +238,104 @@ function randomRange(min: number, max: number) {
   }
   return returnStr;
 }
+
+// test for milestone3
+describe("B64Service", () => {
+  let service: B64Service;
+  let httpClientSpy: jasmine.SpyObj<HttpClient>;
+  let httpClient: HttpClient;
+  let httpTestingController: HttpTestingController;
+  beforeEach(() => {
+    TestBed.configureTestingModule({ imports: [HttpClientTestingModule] });
+    httpClient = TestBed.inject(HttpClient);
+    httpTestingController = TestBed.inject(HttpTestingController);
+    service = TestBed.inject(B64Service);
+  });
+
+  it("should be created", () => {
+    expect(service).toBeTruthy();
+  });
+
+  it("should turn utf8 to b64 and back", () => {
+    let testUTF8 = `󳬏ۓ脶򍫐䏷򻱚1󔊣򧰗J鷍ˑ⨘󝳗ʳꟴ󆋔=є򻥼Ӳ򦿴¦槩7}摠꾀𴮣۝م𬷊
+    ,^⒆!טФ񤨷Յe󫱷ъ"Ҁ*=ߋ󻅷񌍖_ᾀ\ꡝ񲁿g"՝MU񔡆ЀL
+    劆֒񦘰ˑ{坋𹔸lǼc&񓱬񊄸Ӽ:󌈅=̹ɽ渭觙􈯶൰ȣ¡圤𹷟򱄢揋ﺝ􊃻^
+    E̶򱩀򑪟eٌϐ򔜮霗燨综*􍪻񭚴oꕃ𷴨2ҽT񺆥uR혙񗊭:򉪼񙏺ۤƓ騡
+    񱺡􌡩PȌ񸛍񩍢􅓴冖㌃ۄ𦄜7Ž⇆*zѱ澁nަvۭË́듍JҢ䆹M󩗟繶`;
+    expect(service.b64_to_utf8(service.utf8_to_b64(testUTF8))).toEqual(
+      testUTF8
+    );
+  });
+
+  // the situation when there is only one word'test'
+  it("should b63service works when with one word of translation", () => {
+    let testxml = `PFRFST4KICAgIDx0ZXh0IHhtbDpsYW5nPSJlbmciIGZhbGxiYWNrLWxhbmdzPSJ1
+    bmQiIGlkPSJ0MCI+CiAgICAgICAgPGJvZHkgaWQ9InQwYjAiPgogICAgICAgICAgICA8ZGl2IHR5cGU9
+    InBhZ2UiIGlkPSJ0MGIwZDAiPgogICAgICAgICAgICAgICAgPHAgaWQ9InQwYjBkMHAwIj4KICAgICAg
+    ICAgICAgICAgICAgICA8cyBpZD0idDBiMGQwcDBzMCI+PHcgaWQ9InQwYjBkMHAwczB3MCIgQVJQQUJF
+    VD0iVCBFSCBTIFQiPnRlc3Q8L3c+IDx3IGlkPSJ0MGIwZDBwMHMwdzEiIEFSUEFCRVQ9IlQgRUggUyBU
+    Ij50ZXN0PC93Pjwvcz48cCBjbGFzcz0idHJhbnNsYXRpb24iPnpoZXNoaW48L3A+CiAgICAgICAgICAg
+    ICAgICA8L3A+CiAgICAgICAgICAgIDwvZGl2PgogICAgICAgIDwvYm9keT4KICAgIDwvdGV4dD4KPC9URUk+`;
+    let translation = service.b64_to_utf8(testxml);
+    expect(service.b64_to_utf8(service.utf8_to_b64(testxml))).toEqual(testxml);
+  });
+
+  // the situation when the sentence ocupies half of the line
+  it("should b63service works with a medium length translation", () => {
+    let testxml = `PFRFST4KICAgIDx0ZXh0IHhtbDpsYW5nPSJlbmciIGZhbGxiYWNrLWxhbmdzPSJ1bmQi
+    IGlkPSJ0MCI+CiAgICAgICAgPGJvZHkgaWQ9InQwYjAiPgogICAgICAgICAgICA8ZGl2IHR5cGU9InBhZ2Ui
+    IGlkPSJ0MGIwZDAiPgogICAgICAgICAgICAgICAgPHAgaWQ9InQwYjBkMHAwIj4KICAgICAgICAgICAgICAg
+    ICAgICA8cyBpZD0idDBiMGQwcDBzMCI+PHcgaWQ9InQwYjBkMHAwczB3MCIgQVJQQUJFVD0iREggSUggUyI+
+    dGhpczwvdz4gPHcgaWQ9InQwYjBkMHAwczB3MSIgQVJQQUJFVD0iSUggWiI+aXM8L3c+IDx3IGlkPSJ0MGIw
+    ZDBwMHMwdzIiIEFSUEFCRVQ9IkFIIj5hPC93PiA8dyBpZD0idDBiMGQwcDBzMHczIiBBUlBBQkVUPSJUIEVI
+    IFMgVCI+dGVzdDwvdz4gPHcgaWQ9InQwYjBkMHAwczB3NCIgQVJQQUJFVD0iRiBBTyBSIj5mb3I8L3c+IDx3
+    IGlkPSJ0MGIwZDBwMHMwdzUiIEFSUEFCRVQ9Ik0gQVkgTCBTIFQgT1cgTiI+bWlsZXN0b25lPC93PiA8dyBp
+    ZD0idDBiMGQwcDBzMHc2IiBBUlBBQkVUPSJHIFVIIEQiPmdvb2Q8L3c+Ljwvcz48cCBjbGFzcz0idHJhbnNs
+    YXRpb24iPui/meaYr+S4gOS4qua1i+ivlTwvcD4KICAgICAgICAgICAgICAgIDwvcD4KICAgICAgICAgICAg
+    PC9kaXY+CiAgICAgICAgPC9ib2R5PgogICAgPC90ZXh0Pgo8L1RFST4=`;
+    let translation = service.b64_to_utf8(testxml);
```

Comment:
unused variable

Reply:
Thanks. delete the duplicate ones. Fixed